python/lib/core/dmod/core/context/README.md

```diff
@@ -0,0 +1,191 @@
+collections like lists and dictionaries, and keeps a scope manager that will keep items alive until told otherwise.
+
+To use a `DMODObjectManager`'s scope, a scope must first be established. Given the instantiation:
+
+```python
+manager = DMODObjectManager()
+```
+
+call the following to create a new scope:
+
+```python
+import uuid
+
+scope_id = uuid.uuid1()
+new_scope: ObjectManagerScope = manager.establish_scope(scope_id)
+```
+
+This will create an internal scope bound to the `uuid` named `scope_id`. Now, the following can be called to create 
+objects within that scope:
+
+```python
+example: SomeExample = manager.create_object("SomeExample", scope_id)
+example2: SomeExample = new_scope.create_object("SomeExample")
+```
+
+Both commands will create an object tied to the given scope. Calling:
+
+```python
+manager.free(scope_id)
+```
+
+will reduce the reference count on all objects in the scope. If nothing else is referencing the objects within the 
+scope, those objects will be destroyed. When the last of the living proxies that existed within that scope fall out of 
+use (such as by returning from a function that used it), those objects will be destroyed. Operations tied to other 
+processes may also be destroyed upon process completion by using the `DMODObjectManager`'s monitoring functionality.
+
+Given:
+
+```python
+from concurrent.futures import ProcessPoolExecutor
+
+def do_something_with_example(example: SomeExample):
+    ...
+
+executor = ProcessPoolExecutor()
+
+future_result = executor.submit(do_something_with_example, example2)
+
+new_scope.monitor(future_result)
+```
+
+Everything attached to `new_scope` will be freed when `future_result` is complete.
+
+The `DMODObjectManager` also uses its own `Server` implementation. The new implementation supports the use of 
+features such as `properties`, which are _not_ usable in our minimum python versions.
+
+### dmod.core.context.monitor
+
+Defines `FutureMonitor`. This is the default monitor used on `DMODObjectManager` that will signal when the objects 
+within a scope may be removed. Users of the `DMODObjectManager` should be largely shielded from needing to know 
+anything about the monitor. The monitor keeps track of a thread that polls a queue of futures. If a retrieved future 
+is complete, it ends the scope of the associated scope object. The future is thrown back into the queue if it is not 
+yet complete.
+
+### dmod.core.context.proxy
+
+Defines how Proxy objects that will be used with the object managers should be created. Unless told otherwise, proxies 
+will be automatically generated. These generated proxies are also slightly more robust when compared to the 
+automatically generated proxies generated by built in functionality on the versions of Python where managers operate as 
+expected (3.9+). Just about **all** functions, methods, and properties will be usable whereas that's not the case with 
```

Comment:
Autoproxies in <3.9 are outright broken and the fix isn't implemented because it was written too late to be backported. That was touched on earlier in the documentation when talking about the Server. Similarly, _only_ instance methods may be used through vanilla proxies where that's not always reasonable when working like you are _not_ using a proxy. I'm adding more details to the document.

Reply:
Should we consider dropping support for 3.8 due to this?  Its EOL is coming in a few months anyway; I don't think it would be a problem if we dropped it slightly earlier than that if it would save us a lot of trouble (though it may be too late for that).

Reply:
> Similarly, only instance methods may be used through vanilla proxies where that's not always reasonable when working like you are not using a proxy. I'm adding more details to the document.

AutoProxies in <3.9 are broken, but they are very limited in 3.9+. The functionality included in this repo allows you to actually use the proxies in the spirit of the pattern instead of just routing certain functions. From a programming perspective, no real design considerations need to be taken into account when using a proxy vs not using a proxy. The functionality is also very well tested.